lib/sinatra/namespace.rb

```diff
@@ -187,9 +188,11 @@ def namespace_errors
       end
 
       def error(*codes, &block)
-        args  = Sinatra::Base.send(:compile!, "ERROR", /^#{@pattern}/, block)
-        codes = codes.map { |c| Array(c) }.flatten
+        args  = Sinatra::Base.send(:compile!, "ERROR", regexpify(@pattern), block)
```

Comment:
What refactor do you mean? The `regexpify(@pattern)` is the fix for the bug described.